File: DataUtilities_python/Load_XDS.py
# -*- coding: utf-8 -*-

import glob
import os
from xds import lab_data
        
#%% Load XDS
def Load_XDS(Monkey, Date, Task, Morn_vs_Noon):

    ## Define the file location
    base_path = 'C:/Users/rhpow/Documents/Work/Northwestern/Monkey_Data/' + Monkey + '/' + Date + '/'
    if Morn_vs_Noon == 'Trimmed':
        base_path = base_path + 'Trimmed/'
    
    xds_dir = glob.glob(base_path + '*.mat')
    
    # Find the names of each file
    if len(xds_dir) > 1:
        file_names = [[] for ii in range(len(xds_dir))]
        for ii in range(len(xds_dir)):
            file_names[ii] = os.path.basename(xds_dir[ii])
        # Only look at the selected task
        xds_names = list(filter(lambda x: Task in x, file_names))
    else:
        file_names = os.path.basename(xds_dir[0])
        # Only look at the selected task
        xds_names = file_names
        file_name = xds_names

    # Identify the morning or afternoon file
    if Morn_vs_Noon == 'Morn':
        file_name = list(filter(lambda x: 'Pre' in x, xds_names))[0]
    if Morn_vs_Noon == 'Noon':
        file_name = list(filter(lambda x: 'Post' in x, xds_names))[0]
    if isinstance(Morn_vs_Noon, int):
        if len(xds_dir) > 1:
            file_name = xds_names[Morn_vs_Noon]
        else:
            file_name = xds_names
    
    # Display the drug used
    if 'Lex' in file_name:
        print('Drug Used: Lexapro')
    if 'Cyp' in file_name:
        print('Drug Used: Cyproheptadine')
    if 'Caff' in file_name:
        print('Drug Used: Caffeine')
    if 'Con' in file_name:
        print('Drug Used: Control')
        
    # Load the xds file
    
    xds = lab_data(base_path, file_name)
    
    #%% Return the xds file
    
    return xds



















        
        
        

File: DataUtilities_matlab/Load_XDS.m
function [xds] = Load_XDS(Monkey, Date, Task, Sorted, File_Tag)

%% Define the file location
base_xds_dir = strcat('C:\Users\rhpow\Documents\Work\Northwestern\Monkey_Data\', Monkey, '\', Date, '\');

xds_dir = dir(strcat(base_xds_dir, '*.mat'));

% Find the names of each file
xds_names = strings;
if length(xds_dir) > 1
    for ii = 1:length(xds_dir)
        xds_names{ii,1} = xds_dir(ii).name;
    end
else
    xds_names{1,1} = xds_dir.name;
end

% Only look at the selected task
task_idx = contains(xds_names, Task);
xds_names = xds_names(task_idx);

% Sorted vs unsorted
if Sorted == 1
    sorted_idxs = or(contains(xds_names, '-s'), contains(xds_names, '-ms'));
    xds_names = xds_names(sorted_idxs);
else
    unsorted_idxs = and(~contains(xds_names, '-s'), ~contains(xds_names, '-ms'));
    xds_names = xds_names(unsorted_idxs);
end

% Identify the morning or afternoon file
if ischar(File_Tag)
    if strcmp(File_Tag, 'Morn') || strcmp(File_Tag, 'Pre')
        xds_idx = find(contains(xds_names, 'Pre'));
    elseif strcmp(File_Tag, 'Noon') || strcmp(File_Tag, 'Post')
        xds_idx = find(contains(xds_names, 'Post'));
    else
        File_suffix = strings;
        for ii = 1:length(xds_names)
            File_suffix{ii,1} = extractAfter(xds_names{ii,1}, Task);
        end
        xds_idx = find(contains(File_suffix, File_Tag));
        end
elseif isnumeric(File_Tag)
    File_suffix = strings;
    for ii = 1:length(xds_names)
        File_suffix{ii,1} = extractAfter(xds_names{ii,1}, Task);
    end
    xds_idx = find(contains(File_suffix, num2str(File_Tag)));
end
try
    file_name = xds_names{xds_idx};
catch
    xds = NaN;
    disp('File Could Not Be Found')
    return
end

% Display the drug used
if contains(file_name, 'Lex')
    disp('Drug Used: Lexapro')
elseif contains(file_name, 'Cyp')
    disp('Drug Used: Cyproheptadine')
elseif contains(file_name, 'Caff')
    disp('Drug Used: Caffeine')
elseif contains(file_name, 'Con')
    disp('Drug Used: Control')
end

% Load the xds file
load(strcat(base_xds_dir, file_name), 'xds')







